kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -167,6 +167,9 @@ State state() {
 
     public void outboundChannelActive(ChannelHandlerContext ctx) {
         this.outboundCtx = ctx;
+        if (this.state == State.CONNECTED) {
```

Comment:
Arguably this implies that we are missing a state for TLS negotiation in the connection state machine.